pkg/clusterd/context.go

```diff
@@ -22,10 +22,15 @@ import (
 	"github.com/rook/rook/pkg/util/sys"
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
-// The context for loading or applying the configuration state of a service.
+// Context for loading or applying the configuration state of a service.
 type Context struct {
+
+	// The kubernetes config used for this context
+	KubeConfig *rest.Config
```

Comment:
Where is this used? After the clientsets are loaded I didn't think the config was needed anymore.

Reply:
You're right - it's not used in this PR - I used it in code that is not part of this PR, to instantiate a controller-runtime manager like the operator-sdk scaffold is creating:
https://github.com/operator-framework/operator-sdk/blob/9cf300dd1a08341f89ce3b2ee995084b314c5bdf/internal/pkg/scaffold/cmd.go#L107-L128

I want to make another PR to add the dependency on controller-runtime because it seems like it provides good tools to create controllers and is the default dependency of operator-sdk: https://godoc.org/sigs.k8s.io/controller-runtime

Anyway, regarding the config in Context - I could also call the `rook.GetKubeConfig()` again when needed, but thought that it makes sense to keep the resolved config so that any type of client can be instantiated later on if needed and use the config as decided by the context.

Reply:
On second thought, I'll just remove it for now. 
Will add if needed later.

Reply:
For #3067 this may actually helpful so how about we keep it?

Reply:
Sure. I didn't remove it yet. It's also harmless in any case so keeping it. thanks